spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -267,12 +264,9 @@ def _measure_volume(self, im_data, p_lst, idx):
         """
         Measure the volume of the lesion
         """
-        for zz in range(im_data.shape[2]):
-            self.volumes[zz, idx - 1] = np.sum(im_data[:, :, zz]) * p_lst[0] * p_lst[1] * p_lst[2]
-
-        vol_tot_cur = np.sum(self.volumes[:, idx - 1])
-        self.measure_pd.loc[idx, 'volume [mm3]'] = vol_tot_cur
-        printv('  Volume : ' + str(np.round(vol_tot_cur, 2)) + ' mm^3', self.verbose, type='info')
```

Comment:
The only reason I can think of for why the refactor might not be a 1:1 analog is if f `_measure_volume` was called multiple times, causing `self.volumes` to be repeatedly mutated, with the total volume sum changing across calls. (The variable was called `vol_tot_cur`, too, implying that the "current" sum may evolve over time, too.)

However, while it's true that `_measure_volume` does seem to be called [within a loop](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/02d775aabed88c5d871cf087ab7cfe8f06578acb/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L457-L468), all of the changes to the input arguments at the caller site are dependent on `idx`, and the volume sum is dependent on `idx` too. 

So, I think for the old code, each individual sum ended up completely isolated between calls, despite multiple calls sharing the same object? Meaning that eliminating `self.volumes` is definitely the way to go. :)

Reply:
> So, I think for the old code, each individual sum ended up completely isolated between calls, despite multiple calls sharing the same object?

That's my understanding, yes. The `vol_tot_cur` is only the sum for for a specific `idx`, namely the one that the function has just completely overwritten.